src/cljs/main/broadfcui/common/links.cljs

```diff
@@ -11,6 +11,17 @@
                         attributes)
    contents])
 
+(defn create-download [label url filename]
```

Comment:
If you change the signature to `[label url & [filename]]` this will work for external links, too.

Reply:
By external do you mean... files hosted outside of FireCloud? Why would `filename` be optional in that case?

Or do you want to create a new function called "create any link you want" that takes all the possible attributes, and replaces the bodies of `create-external` and `create-download`?

Reply:
discussed in person, implemented by making the function multi-arity